modules/gateways/monero/createinvoice.php

```diff
@@ -16,15 +16,17 @@
 $link = $GATEWAY['daemon_host'].":".$GATEWAY['daemon_port']."/json_rpc";
 
 
-function monero_payment_id(){
-    if(!isset($_COOKIE['payment_id'])) { 
+
+function monero_payment_id($invoice_id){
+	$invoice_id = checkCbInvoiceID($invoice_id, "monero");
+   	if(!isset($_COOKIE["payment_id$invoice_id"])) { 
 		$payment_id  = bin2hex(openssl_random_pseudo_bytes(8));
-		setcookie('payment_id', $payment_id, time()+2700);
+		// create cookie per invoice_id.
+		setcookie("payment_id$invoice_id", $payment_id, time()+2700);
```

Comment:
I can't understand the purpose of this change. Why should you call a cookie with the invoice id?